doc/overlays.xml

```diff
@@ -0,0 +1,99 @@
+<chapter xmlns="http://docbook.org/ns/docbook"
+         xmlns:xlink="http://www.w3.org/1999/xlink"
+         xml:id="chap-overlays">
+
+<title>Overlays</title>
+
+<para>This chapter describes how to extend and change Nixpkgs content using
+overlays.  Overlays are used to add phases in the fix-point used by Nixpkgs
+to bind the dependencies of all packages.</para>
+
+<!--============================================================-->
+
+<section xml:id="sec-overlays-install">
+<title>Installing Overlays</title>
+
+<para>Overlays are looked for in the following order, the first valid one is
+considered, and all the rest are ignored:
```

Comment:
What I meant is the fallback logic, we consider the environment variable `NIXPKGS_OVERLAYS`, and check if this is a directory.  Then we do the same check for the `$HOME/.nixpkgs/overlays`.

I agree, we should throw if `NIXPKGS_OVERLAYS` is set, but the content does not validate.
But I do not think we should do anything if the `$HOME/.nixpkgs/overlays` directory does not exists.